src/lib.rs

```diff
@@ -381,6 +381,64 @@ impl Iterator for PRegClass {
             Some(PReg::from(reg_index))
         }
     }
+
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        if n >= self.len() {
+            self.regs = 0;
+            None
+        } else {
+            let n_from_right = self.len() - n;
+            // this is the number of trailing zeros for the n-th set bit from the right
+            let index = find_nth(self.regs, n_from_right as u64);
+            // clear those bits
+            self.regs &= u64::MAX << index;
+            self.regs &= !(1u64 << index);
+            // calculate the PReg
+            let reg_index = index as u8 | self.class_mask;
+            Some(PReg::from(reg_index))
+        }
+    }
+}
+
+// find the n-th set bit from the RIGHT,
+// using 1 based indexing
+// returns distance from the LEFT
+fn find_nth(v: u64, mut r: u64) -> u64 {
+    const C: u64 = 0b00000000_11111111_00000000_11111111_00000000_11111111_00000000_11111111; // 0x00FF00FF
+    const D: u64 = 0b00001111_00001111_00001111_00001111_00001111_00001111_00001111_00001111; // 0xF0F0F0F0
+    const E: u64 = 0b00110011_00110011_00110011_00110011_00110011_00110011_00110011_00110011; // 0x33333333
+    const F: u64 = 0b01010101_01010101_01010101_01010101_01010101_01010101_01010101_01010101; // 0x55555555
+
+    // from https://graphics.stanford.edu/~seander/bithacks.html: uses 1 based indexing
```

Comment:
Which of the algorithms on that page is this?

Reply:
Select the bit position (from the most-significant bit) with the given count (rank) 
Unfortunately the links to the individual algorithms don't seem to work

Reply:
https://graphics.stanford.edu/~seander/bithacks.html##SelectPosFromMSBRank works. This page (likely accidentally) used `#SelectPosFromMSBRank` as name rather than `SelectPosFromMSBRank` so you need two hashes.

Reply:
Thanks, changed the link